### PR TITLE
update ts configs

### PR DESCRIPTION
**Pull request by TomWoodward**

### CODE REVIEW

File: src/Highlight.ts [hunk 1]
```diff
@@ -17,11 +17,13 @@ export default class Highlight {
...
       this._range = arg;
       this._content = arg2;
-    } else {
+    } else if (content) {
       this._id = arg;
       this._range = arg2;
       this._content = content;
     }
+
+    throw new Error('invalid constructor arguments');
```
Comment (philschatz) [inline src/Highlight.ts]: Doesn't this cause the constructor to always throw an Error? Maybe I missed something by just viewing the diff.

File: src/Highlighter.test.ts [hunk 2]
```diff
@@ -14,6 +14,10 @@ describe('Reference elements', () => {
     const container = document.getElementById('container');
     const reference = document.getElementById('referenceElement2');
 
+    if (!container) {
+      throw new Error('can\'t find container');
```
Comment (philschatz) [inline src/Highlighter.test.ts]:
```suggestion
      throw new Error('cannot find container');
```

File: src/Highlighter.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ interface IOptions {
   snapMathJax?: boolean;
   snapWords?: boolean;
   className?: string;
-  onClick?: (highlight?: Highlight) => void;
+  onClick?: (highlight?: Highlight | null) => void;
```
Comment (philschatz) [inline src/Highlighter.ts]: Does this method need to handle all 3 cases? `onClick(highlight)`, `onClick()`, and `onClick(null)`? It seems a little confusing. From the references to `onClick` in this file it does not seem like `highlight` would ever be null... when would that occur?

Also, if `highlight` could be null, would it be worth creating an Optional type for readability? Example:

```ts
export type Optional<T> = T | null | undefined
```

**Comment by TomWoodward** [comment]
@philschatz made some updates for you

**Pull request merged**